### PR TITLE
DateMath: Expand Triggers

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
Add a bunch more common forms that I've seen in the searches - also
supports a couple of forms that should've been supported before.

Things like `What day will it be in 120 days?`, `What time was it 3 hours ago?` etc...

---
IA Page: https://duck.co/ia/view/date_math

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by GuiltyDolphin** [comment]
@mintsoft I've played around with this on Beta and it looks good as far as I can tell. What do you think?

**Comment by GuiltyDolphin** [comment]
@moollaza @mintsoft Ping!

**Comment by mintsoft** [comment]
@GuiltyDolphin Apologies; I totally forgot about this. I'll take a look now

File: lib/DDG/Goodie/DateMath.pm [hunk 1]
```diff
@@ -79,6 +79,9 @@ sub build_result {
         return $result, structured_answer => {
             id   => 'date_math',
             name => 'Answer',
+            meta => {
+                signal => 'high',
```
Comment (mintsoft) [inline lib/DDG/Goodie/DateMath.pm]: Is this clashing with calculator or something?

File: lib/DDG/Goodie/DateMath.pm [hunk 2]
```diff
@@ -27,8 +27,8 @@ sub get_duration {
 
 sub get_action_for {
     my $action = shift;
-    return '+' if $action =~ /\+|plus|from/i;
-    return '-' if $action =~ /\-|minus|ago/i;
+    return '+' if $action =~ /^\+|plus|from|in$/i;
```
Comment (mintsoft) [inline lib/DDG/Goodie/DateMath.pm]: maybe add "add" and "subtract" ?

File: lib/DDG/Goodie/DateMath.pm (hunk 1 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/DateMath.pm]: Ugh... Did I not do an example query? No, it was clashing with lyrics or something... [And news apparently](https://duckduckgo.com/?q=3+hours+ago&t=ffab&ia=news)!

File: lib/DDG/Goodie/DateMath.pm (hunk 2 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/DateMath.pm]: Sure.

**Comment by GuiltyDolphin** [comment]
@mintsoft Done. Anything else?

**Comment by mintsoft** [comment]
@GuiltyDolphin LGTM

@moollaza any issues with the `signal => 'high'` being added here?

**Comment by GuiltyDolphin** [comment]
@mintsoft @moollaza It shouldn't be an issue - given the specific queries that trigger the IA. I haven't run across any situations on Beta.

**Comment by moollaza** [comment]
This LGTM :+1:

> @moollaza any issues with the signal => 'high' being added here?

Nope :+1:

**Pull request merged**